cmd/livepeer/livepeer.go

```diff
@@ -330,6 +341,49 @@ func main() {
 		// Setup unbonding service to manage unbonding locks
 		n.EthServices["UnbondingService"] = eventservices.NewUnbondingService(n.Eth, dbh)
 
+		// Initialize block watcher that will emit logs used by event watchers
+		blockWatcherClient, err := blockwatch.NewRPCClient(*ethUrl, ethRPCTimeout)
+		if err != nil {
+			glog.Errorf("Failed to setup blockwatch client: %v", err)
+			return
+		}
+		topics := watchers.FilterTopics()
+		blockWatcherCfg := blockwatch.Config{
+			Store:               n.Database,
+			PollingInterval:     blockWatcherPollingInterval,
+			StartBlockDepth:     rpc.LatestBlockNumber,
+			BlockRetentionLimit: blockWatcherRetentionLimit,
+			WithLogs:            true,
+			Topics:              topics,
+			Client:              blockWatcherClient,
+		}
+		// Wait until all event watchers have been initialized before starting the block watcher
+		blockWatcher := blockwatch.New(blockWatcherCfg)
+
+		blockWatchCtx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		// Backfill events that the node has missed since its last seen block. This method will block
+		// and the node will not continue setup until it finishes
+		if err := blockWatcher.BackfillEventsIfNeeded(blockWatchCtx); err != nil {
+			glog.Errorf("Failed to backfill events: %v", err)
+			return
+		}
+
+		go func() {
+			if err := blockWatcher.Watch(blockWatchCtx); err != nil {
+				glog.Errorf("block watcher error: %v", err)
+			}
+		}()
+
+		roundsWatcher, err := watchers.NewRoundsWatcher(blockWatcher, n.Eth, addrMap["RoundsManager"])
+		if err != nil {
+			glog.Errorf("Failed to setup roundswatcher: %v", err)
+			return
+		}
+		go roundsWatcher.Watch()
```

Comment:
Let's start the round watcher before starting the block watcher (and generally we can start event watchers before the block watcher) to ensure that the round watcher is ready when the block watcher starts emitting blocks.

Reply:
fixed in 5399d9d